master/buildbot/process/build.py

```diff
@@ -325,7 +325,10 @@ def startBuild(self, build_status, workerforbuilder):
         # If it returns failure then we don't start a new build.
         if ready_or_failure is not True:
             yield self.buildPreparationFailure(ready_or_failure, "worker_prepare")
-            self.buildFinished(["worker", "not", "available"], RETRY)
```

Comment:
AbstractLatentWorker already has worker missing timer to cover the start timeout.

AbstractLatentWorker.substantiate() will start a missing_timer before calling create_instance(), when it is fired, it will call AbstractLatentWorker._missing_timer_fired which will notify _substantiation_notifier as Failure, so here the workerForBuilder.prepare() will have a Failure exception.